GeneratorInterface/LHEInterface/src/LHEEvent.cc

```diff
@@ -306,7 +306,7 @@ std::auto_ptr<HepMC::GenEvent> LHEEvent::asHepMCEvent() const
 			genVertices.push_back(current_vtx);
 		}
 
-		for(unsigned int j = mother1; j <= mother2; j++)	// set mother-daughter relations
+		for(unsigned int j = mother1; j <= mother2; j++) {	// set mother-daughter relations
 			if (!genParticles.at(j)->end_vertex())
 				current_vtx->add_particle_in(genParticles.at(j));
 
```

Comment:
I don't think this is right. With your new code, we get `mother2-mother1+1` copies of particle `i` added to the current `current_vtx`. I'm pretty sure we only want one which is what the original code would do.
The original code would only proceed if `mother1` != 0 which is probably sensible since we are translating from FORTRAN and the number there would start at 1.